shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt

```diff
@@ -45,11 +44,16 @@ class MealApi {
         var meals: List<Meal> = getAllMeals()
         for (meal in meals) {
             meal.distance = locationUtil.getDistance(
-                userLat, userLon, meal.locationLat.toDouble(), meal.locationLong.toDouble(), distanceUnit
+                userLat,
+                userLon,
+                meal.locationLat.toDouble(),
+                meal.locationLong.toDouble(),
+                distanceUnit
             )
         }
-        // TODO: Filter meals that have expired here
-        return meals.sortedWith(compareBy { it.distance })
+        return meals
+            .filter { SharedDate().isBefore(SharedDate(it.expiryDate.toLong())) }
```

Comment:
filter meals where the date now, is before the expiry date.

I think this is easier to understand now